Code/GraphMol/WedgeBonds.cpp

```diff
@@ -0,0 +1,494 @@
+}
+}  // namespace
+
+namespace detail {
+
+std::pair<bool, INT_VECT> countChiralNbrs(const ROMol &mol, int noNbrs) {
+  // we need ring information; make sure findSSSR has been called before
+  // if not call now
+  if (!mol.getRingInfo()->isInitialized()) {
+    MolOps::findSSSR(mol);
+  }
+
+  INT_VECT nChiralNbrs(mol.getNumAtoms(), noNbrs);
+
+  // start by looking for bonds that are already wedged
+  for (const auto bond : mol.bonds()) {
+    if (bond->getBondDir() == Bond::BEGINWEDGE ||
+        bond->getBondDir() == Bond::BEGINDASH ||
+        bond->getBondDir() == Bond::UNKNOWN) {
+      if (bond->getBeginAtom()->getChiralTag() == Atom::CHI_TETRAHEDRAL_CW ||
+          bond->getBeginAtom()->getChiralTag() == Atom::CHI_TETRAHEDRAL_CCW) {
+        nChiralNbrs[bond->getBeginAtomIdx()] = noNbrs + 1;
+      } else if (bond->getEndAtom()->getChiralTag() ==
+                     Atom::CHI_TETRAHEDRAL_CW ||
+                 bond->getEndAtom()->getChiralTag() ==
+                     Atom::CHI_TETRAHEDRAL_CCW) {
+        nChiralNbrs[bond->getEndAtomIdx()] = noNbrs + 1;
+      }
+    }
+  }
+
+  // now rank atoms by the number of chiral neighbors or Hs they have:
+  bool chiNbrs = false;
+  for (const auto at : mol.atoms()) {
+    if (nChiralNbrs[at->getIdx()] > noNbrs) {
+      // std::cerr << " SKIPPING1: " << at->getIdx() << std::endl;
+      continue;
+    }
+    auto type = at->getChiralTag();
+    if (type != Atom::CHI_TETRAHEDRAL_CW && type != Atom::CHI_TETRAHEDRAL_CCW) {
+      continue;
+    }
+    nChiralNbrs[at->getIdx()] = 0;
+    chiNbrs = true;
+    for (const auto nat : mol.atomNeighbors(at)) {
+      if (nat->getAtomicNum() == 1) {
+        // special case: it's an H... we weight these especially high:
+        nChiralNbrs[at->getIdx()] -= 10;
+        continue;
+      }
+      type = nat->getChiralTag();
+      if (type != Atom::CHI_TETRAHEDRAL_CW &&
+          type != Atom::CHI_TETRAHEDRAL_CCW) {
+        continue;
+      }
+      nChiralNbrs[at->getIdx()] -= 1;
+    }
+  }
+  return std::make_pair(chiNbrs, nChiralNbrs);
+}
+
+//
+// Determine bond wedge state
+///
+Bond::BondDir determineBondWedgeState(const Bond *bond,
+                                      unsigned int fromAtomIdx,
+                                      const Conformer *conf) {
+  PRECONDITION(bond, "no bond");
+  PRECONDITION(bond->getBondType() == Bond::SINGLE,
+               "bad bond order for wedging");
+  const auto mol = &(bond->getOwningMol());
+  PRECONDITION(mol, "no mol");
+
+  auto res = bond->getBondDir();
+  if (!conf) {
+    return res;
+  }
+
+  Atom *atom;
+  Atom *bondAtom;
+  if (bond->getBeginAtom()->getIdx() == fromAtomIdx) {
+    atom = bond->getBeginAtom();
+    bondAtom = bond->getEndAtom();
+  } else {
+    atom = bond->getEndAtom();
+    bondAtom = bond->getBeginAtom();
+  }
+
+  auto chiralType = atom->getChiralTag();
+  TEST_ASSERT(chiralType == Atom::CHI_TETRAHEDRAL_CW ||
```

Comment:
I might consider this a precondition that the fromAtom is chiral.

Reply:
I can change this from `TEST_ASSERT()` to `PRECONDITION()`, but the test more or less needs to stay here since it requires identifying the relevant atom. 
The effect in the end will be exactly the same aside from the message which is generated

Reply:
How likely is this assert to happen?  I'm assuming Bond has already been checked to have a chiral atom?

Reply:
This isn’t new code. It was just moved from MolFileStereochem.cpp
I think it’s low risk